doc/developer/design/20220516_transactional_consistency.md

```diff
@@ -0,0 +1,449 @@
+- Transaction T1 is ordered before Transaction T2 if the timestamp of T1 == the timestamp of T2, T1 is write-only, and
+  T2 is read-only.
+- Transaction T1 can be arbitrarily ordered with Transaction T2 if the timestamp of T1 == the timestamp of T2, T1 is
+  read-only, and T2 is read-only.
+
+#### Strict Serializable
+
+See [https://jepsen.io/consistency/models/strict-serializable](https://jepsen.io/consistency/models/strict-serializable)
+for an in depth formal discussion of Strict Serializable. Strict Serializable is a combination of Serializable and
+Linearizable, which means that the total order formed by all transactions must be constrained by their real time
+occurrences. So if transaction t1 happened in real time before transaction t2, then t1 must be ordered before t2 in the
+total ordering.
+
+Operations that only include user tables are currently Strict Serializable. This is made easy by using a single global
+timestamp and a single thread in the coordinator to order operations. Writes happen at a timestamp larger than all
+previous operations. Reads happen at a timestamp equal to or larger than all previous operations.
+
+Operations that involve reading from views over upstream data are not currently Strict Serializable. It is possible to
+read from a view at one timestamp and then later read from the same view at an earlier timestamp. Consider the following
+scenario:
+
+1. Client reads from view v1 that has an upper of `ts + 1` and a since of `ts - 10`. The read will happen at `ts`.
+2. Client reads from a join over v1 and view v2. v2 has an upper of `ts - 5` and some since less or equal to `ts - 6`.
+   The read will happen at `ts - 6`.
+
+Reference [Timestamp Selection](#Timestamp Selection) to see why these timestamps are selected. This is an obvious
+violation of strict serializability because the second read happens later than the first read in real time, but it is
+assigned an earlier timestamp.
+
+In order to achieve Strict Serializability in the general case, Materialize must always select a timestamp greater than
+all previously selected timestamps. This will ensure that if an operation happens later in real time, then it will be
+assigned a later timestamp. Consecutive reads can all be assigned an equal timestamp as long as there’s no writes in
+between them. This is made somewhat easier by the fact that all timestamp selection happens on a single thread so no
+coordination is needed. One implementation for this is to consult the global timestamp for all reads, not just ones
+involving user tables.
+
+If we use the global timestamp for all reads, then the global timestamp will have to advance in the following cases:
+
+1. Client writes to a user table.
+2. One second passes and the global timestamp is below the wall clock.
+3. Client reads from a view that has a [`since`, `upper`) range larger than the current global timestamp.
```

Comment:
yeah this sentence is lazy. I changed it to

> Client reads from a view that requires a timestamp larger than the current global timestamp. Depending on how a
   timestamp is selected for this view, this can occur if `since` is larger than the global timestamp, `upper` is larger
   than the global timestamp, or both are larger than the global timestamp.

Does that clarify it?